### ISSUE TITLE
Fatal Python error: (pygame parachute) Segmentation Fault, and Keras exiting without errors

### ISSUE THREAD

**Issue by Gannio** [HackerPoet/Composer#4]
So the application was working after the fix in Issue #2, however after going back to do a training session, and returing to live_edit, I got this:

`Fatal Python error: (pygame parachute) Segmentation Fault`

by doing some print-debugging, I found the area this error occurs at to be here:

```
import pygame
print "Loading Encoder..."
try:
	print("Load Model")#The line below is where the error occurs and the program terminated. Around line 180 in the original code.
	model = load_model(dir_name + 'model.h5')
	print("enc")
	enc = K.function([model.get_layer('encoder').input, K.learning_phase()],
					 [model.layers[-1].output])
	enc_model = Model(inputs=model.input, outputs=model.get_layer('pre_encoder').output)
```
I tried reinstalling Anaconda and all libraries multiple times to try and correct this, however it still occurs, and now there's also another problem with train.py, with the output only showing 'Loading Keras...' then exiting the entire program after a few seconds, without any error statements.

While doing research online about the former issue, I found that pygame can actually be pretty buggy between versions also, so if it helps I used pipreqs to get my current version layout, in case maybe I'm using a too upgrade or downgraded version of any of these:

```
matplotlib==2.2.2
pygame==1.9.4
pydot==1.2.4
numpy==1.14.3
Keras==2.2.2
mido==1.2.8
Theano==1.0.2
pyaudio==0.2.11
```

**Comment by HackerPoet** [comment]
Keras's `load_model` usually only works when the model was saved with the same version of Keras as the version loading it.  But it should still print an error if something is wrong, not exit unexpectedly...

Sounds more like an issue with either pygame or keras than with the composer.  For reference, here's what I'm using at the moment, it doesn't look too different:
```
numpy==1.14.5
Keras==2.2.0
pydot==1.2.3
Theano==0.9.0
PyAudio==0.2.11
pygame==1.9.3
```

**Comment by Gannio** [comment]
After a few more days of wrangling with it, I finally got it to work. I'll post some advice for what I used to get it going again.

Firstly, I'd recommend Miniconda for setup if you haven't setup python yet. As for the required classes, I'm providing a pip freeze this time since apparently pipreqs doesn't get classes that are required by classes called in scripts:
```

asn1crypto==0.24.0
backports-abc==0.5
backports.functools-lru-cache==1.5
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
conda==4.5.9
cryptography==2.2.2
cycler==0.10.0
docopt==0.6.2
enum34==1.1.6
functools32==3.2.3.post2
futures==3.2.0
h5py==2.8.0
idna==2.6
ipaddress==1.0.22
Keras==2.2.0
Keras-Applications==1.0.2
Keras-Preprocessing==1.0.1
kiwisolver==1.0.1
mako==1.0.7
MarkupSafe==1.0
matplotlib==2.2.2
menuinst==1.4.14
mido==1.2.8
mkl-fft==1.0.5
numpy==1.14.5
pipreqs==0.4.9
PyAudio==0.2.11
pycosat==0.6.3
pycparser==2.18
pydot==1.2.3
pygame==1.9.3
pygpu==0.7.6
pyOpenSSL==18.0.0
pyparsing==2.2.0
PySocks==1.6.8
python-dateutil==2.7.3
pytz==2018.5
pywin32==223
PyYAML==3.13
requests==2.18.4
ruamel-yaml==0.15.37
scipy==1.1.0
singledispatch==3.4.0.3
six==1.11.0
Theano==1.0.2+2.gc449c8699
tornado==5.1
urllib3==1.22
win-inet-pton==1.0.1
wincertstore==0.2
yarg==0.1.9

```
Lastly, make sure you don't have any other CUDA versions on the system (or at the very least, cudnn). I was initially confused by the gpu.theanorc file, and thought I'd need 8.0 for this and 9.1 for deep doodle, but 9.1 will work fine for this (you'll just need to shift the directory in the gpu file).

For those curious, I have a GTX 970M as my graphics card, and I think due to it possibly being older, ran into 'out of memory' problems if the computer hasn't restarted. I've seemed to fix them by adding this line in the gpu file:

```
[gpuarray]

preallocate = 2400

```
Try and get the number close to what it is when it works naturally (for me, 2.4GB, which translates to 2400MB).

The only error I've seen with this is when holding down left click on the command window (Windows 10+ I think), it'll bring up this error, which probably has to do with that stalling command line applications:

`700/942 [=====================>........] - ETA: 0s - loss: 0.0091D:\python\Miniconda2\lib\site-packages\keras\callbacks.py:120: UserWarning: Method on_batch_end() is slow compared to the batch update (8.145000). Check your callbacks.`

Also, slightly off topic, but how many songs did you have for your network?
